service/src/main/java/com/fecklessweasel/service/FileUploadServlet.java

```diff
@@ -0,0 +1,85 @@
+package com.fecklessweasel.service;
+
+import com.fecklessweasel.service.datatier.SQLInteractionInterface;
+import com.fecklessweasel.service.datatier.SQLSource;
+import com.fecklessweasel.service.objectmodel.FileMetadata;
+import com.fecklessweasel.service.objectmodel.ServiceException;
+import com.fecklessweasel.service.objectmodel.UserSession;
+
+import java.io.*;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Date;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+@WebServlet("/servlet/file/upload")
+@MultipartConfig
+public class FileUploadServlet extends HttpServlet {
+
+    private static final String FILEPATH_PREFIX = "/files";
+
+    @Override
+    protected void doPost(final HttpServletRequest request,
+                          final HttpServletResponse response)
+            throws ServletException, IOException {
+
+        // TODO: These should go into database eventually
+        String title = request.getParameter("title");
+        String description = request.getParameter("description");
+
+        Part filePart = request.getPart("file[0]");
+        String fileName = filePart.getSubmittedFileName();
+
+        final UserSession session = UserSessionUtil.resumeSession(request);
+        // If user is not authenticated
+        if (session == null) {
+            return;
+        }
+
+        // Open a SQL connection and create the file meta data.
+        FileMetadata fileMetadata = SQLSource.interact(new SQLInteractionInterface<FileMetadata>() {
+            @Override
+            public FileMetadata run(Connection connection)
+                    throws ServiceException, SQLException {
+
+
+                int classId = Integer.parseInt(request.getParameter("class"));
+                return FileMetadata.create(connection, session.getUser(), classId, new Date());
+            }
+        });
+
+        String filePath = FILEPATH_PREFIX + fileMetadata.getCourse() + fileName;
+        saveFile(filePart.getInputStream(), filePath);
```

Comment:
We don't check for proper file upload. If file upload fails you should throw a ServiceException of some type so that the user is redirected to an error page. Otherwise we are simply assuming that the upload succeeded.